Added trim functionality to Text Column

jamessessford opened the pull request:
Morning folks,

TLDR: Added trim functionality to Text Column

I found that I was using the getValueUsing callback a lot of the time just to trim the length of the content but if I did this I lost the ability to search or sort on that field.

I've added a trim helper to the Text column that takes a trim length and then trims the value after it's been taken from the record.

Example usage:

```php
Columns\Text::make('reference')->primary()->searchable()->sortable()->trim(6),
```

This allows me to only show the first six characters of the reference in the table but retain search and sort functionality.

Cheers!

jamessessford committed:
Added trim functionality to Text Column

danharrin commented:
Hey James! This sounds like a job for `formatUsing` rather than `getValueUsing`. We should use the `trim()` method to set that callback, similar to `date()`. Let me know how that goes.

ryangjchandler commented:
I think this method should probably be called `truncate` or `limit`, since PHP's `trim` function is for removing characters at each end, not actually limiting the length of a string.

jamessessford commented:
Totally agreed @ryangjchandler - Updating now

jamessessford committed:
Changed trim to limit
Changed function to use formatUsing

jamessessford commented:
I've changed the function to limit rather than trim (that was the str helper it's piggybacking) and updated to use formatUsing rather than getValueUsing.

Cheers!

ryangjchandler commented on packages/tables/src/Columns/Text.php:
```diff
@@ -85,4 +86,15 @@ public function options($options)
+    public function limit($length = -1)
+    {
+        $this->formatUsing = function ($value) use ($length) {
+            $value = Str::of($value)->limit($length);
```
No need for the `Str::of()` here - you could just use `Str::limit` directly, right?

jamessessford replied:
You can indeed! I didn't realise you could do that, buzzing I've learned something new this morning :)

jamessessford replied:
I've updated this to:

```php
return Str::limit($value, $length);
```

ryangjchandler replied:
Nice one @jamessessford!

jamessessford committed:
Refactored Str call when limiting string
